### PR TITLE
Ux improvements

### PR DESCRIPTION
**Pull request by fuiszgt**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] PR has a descriptive title
- [x] PR contains closely related changes only (should not have many unrelated changes)
- [x] STEP pod members have been added as reviewers
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the current behavior?
Screenshot from 2020-09-15 17-32-18

Issue Number: N/A

## Other information

New features:
- Indicating task loading with progress bar
- Instructions for the user on a dismissible banner
- Disabling schedule button until tasks are selected
- Indicating scheduling in progress
- Making the list title bold, making the Select-all checkbox unchecked when no tasks are checked
- Fixing the buttons to the bottom

Screenshot 2020-09-15 at 7 13 45 PM - Display 2

- Schedule button becomes enabled after task has been selected
Screenshot 2020-09-15 at 7 14 00 PM - Display 2

- The user can hide the description, and it won't show again on reload
Screenshot 2020-09-15 at 7 14 09 PM - Display 2

- The scheduling is reflected with a progress indicator
Screenshot 2020-09-15 at 7 14 15 PM - Display 2

### CODE REVIEW

**Comment by zlorf** [comment]
Please include before/after screenshots in the PR description

**Comment by fuiszgt** [comment]
> Please include before/after screenshots in the PR description

I just did. :) The version is also deployed.

**Comment by zlorf** [comment]
> > Please include before/after screenshots in the PR description
> 
> I just did. :) The version is also deployed.

Oh wow, several screenshots and detailed bulletpoints! Nice

File: src/main/webapp/script.js [hunk 1]
```diff
@@ -16,21 +16,43 @@
 
 $(document).ready(init);
 
+const taskLoadingProgressBar = new mdc.linearProgress.MDCLinearProgress(document.querySelector('#progress-bar'));
+const scheduleButton = $("#schedule-button");
```
Comment (zlorf) [inline src/main/webapp/script.js:20]: You should not read DOM only before `init`, as the content might not be ready.

File: src/main/webapp/script.js [hunk 2]
```diff
@@ -59,11 +81,20 @@ function renderSingleTask(task) {
     </li>`);
 }
 
+function handleEmptySelection() {
+  isAnyChecked = $(":checkbox[name='taskId']:checked").length > 0;
```
Comment (zlorf) [inline src/main/webapp/script.js:85]: Use `const` before variable name

Comment (zlorf) [reply src/main/webapp/script.js:85]: jQuery docs suggest to use `input:checkbox` instead of just `:checkbox`.

File: src/main/webapp/script.js [hunk 3]
```diff
@@ -59,11 +81,20 @@ function renderSingleTask(task) {
...
+  scheduleButton.prop("disabled", !isAnyChecked);
+  if(!isAnyChecked){
```
Comment (zlorf) [inline src/main/webapp/script.js:87]: nit: add spaces: `if (!isAnyCheked) {`

**Comment by dola** [comment]
For me the "There are no tasks to schedule" message is aligned at the bottom vertically. I think it should be at the top instead.
see https://screenshot.googleplex.com/zDjrjTUqnqQmkPL.png

File: src/main/webapp/WEB-INF/index.html [hunk 4]
```diff
@@ -12,8 +12,19 @@
 <div class="container">
   <div class="tasks-container">
 
+    <div class="banner" id="info">
+      <p class="banner-text"> Hey! Select some tasks you would like to move to your calendar, and hit schedule! You can also select the timerange of scheduling, and set duration for tasks.</p>
```
Comment (dola) [inline src/main/webapp/WEB-INF/index.html:16]: Would make this a bit more concise and e.g. remove the "Hey". Maybe you can get some inspiration from Google's public UX writing guidelines.
https://material.io/design/communication/writing.html#principles

**Comment by dola** [comment]
For me the "There are no tasks to schedule" message is aligned at the bottom vertically. I think it should be at the top instead.
image

File: src/main/webapp/WEB-INF/index.html [hunk 5]
```diff
@@ -12,8 +12,19 @@
...
+      <div class="mdc-touch-target-wrapper banner-button" onclick="hideInfo()">
+        <button class="mdc-button mdc-button--touch">
+          <div class="mdc-button__ripple"></div>
+          <span class="mdc-button__label">GOT IT</span>
```
Comment (dola) [inline src/main/webapp/WEB-INF/index.html:20]: Just use normal casing in HTML and if you want the style to be all uppercase, instead use the CSS property https://developer.mozilla.org/en-US/docs/Web/CSS/text-transform

File: src/main/webapp/style.css [hunk 6]
```diff
@@ -26,6 +27,24 @@ body {
...
 }
 
+.banner{
+  padding: 8px;
+}
+
+.banner-text{
+  line-height: 20px;
+  font-size: 14px;
+  margin:  0px;
```
Comment (dola) [inline src/main/webapp/style.css:37]: nit: if `0` you don't need a unit.

**Comment by Giost** [comment]
After the scheduling, the schedule button is active even if there are no tasks selected

image

**Comment by Giost** [comment]
It also happens if you schedule all tasks in your lists and you have no tasks left, but the schedule button is active

image

File: src/main/webapp/script.js [hunk 7]
```diff
@@ -62,7 +62,7 @@ function renderTasks(tasks) {
 function renderSingleTask(task) {
   id = task.id;
   $("#task-list").append(
-    `<li>
+      `<li>
```
Comment (zlorf) [inline src/main/webapp/script.js:65]: I guess you should revert this small change, as it would cause merge conflicts.

**Comment by zlorf** [comment]
After I schedule some tasks and task list is reloaded, then toggle-all checkbox remain in the same state, that can be checked.
I guess that `renderTasks` should deselect it.

**Review by zlorf — APPROVED** [review]

**Pull request merged**